lib/open_project/text_formatting/filters/autolink_custom_protocols_filter.rb

```diff
@@ -0,0 +1,94 @@
+# frozen_string_literal: true
+
+#-- copyright
+# OpenProject is an open source project management software.
+# Copyright (C) the OpenProject GmbH
+#
+# This program is free software; you can redistribute it and/or
+# modify it under the terms of the GNU General Public License version 3.
+#
+# OpenProject is a fork of ChiliProject, which is a fork of Redmine. The copyright follows:
+# Copyright (C) 2006-2013 Jean-Philippe Lang
+# Copyright (C) 2010-2013 the ChiliProject Team
+#
+# This program is free software; you can redistribute it and/or
+# modify it under the terms of the GNU General Public License
+# as published by the Free Software Foundation; either version 2
+# of the License, or (at your option) any later version.
+#
+# This program is distributed in the hope that it will be useful,
+# but WITHOUT ANY WARRANTY; without even the implied warranty of
+# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+# GNU General Public License for more details.
+#
+# You should have received a copy of the GNU General Public License
+# along with this program; if not, write to the Free Software
+# Foundation, Inc., 51 Franklin Street, Fifth Floor, Boston, MA  02110-1301, USA.
+#
+# See COPYRIGHT and LICENSE files for more details.
+#++
+
+module OpenProject::TextFormatting
+  module Filters
+    # HTML Filter for auto_linking custom protocols in HTML.
+    #
+    # Context options:
+    #
+    #   autolink:
+    #     classes: (string) Classes to add to auto linked urls
+    #     enabled: (boolean)
+    #
+    # This filter does not write additional information to the context.
+    class AutolinkCustomProtocolsFilter < HTML::Pipeline::Filter
+      include ActionView::Helpers::TagHelper
+
+      # Skip text nodes that are within preformatted or linked blocks
+      SKIPPED_BLOCKS = %w(pre code kbd a).to_set
+
+      # Match custom schemes and ignore trailing punctuation
+      def self.regexp
+        %r{((?:#{Setting.allowed_link_protocols.join('|')}):/?/?[^\s<\u00A0"]*[^\s<\u00A0",;\.])}i
```

Comment:
```ruby
pry(main)> Setting.allowed_link_protocols
=> ["f.*x"]
```

And then input

```
fax://foo.bar.test

faaaaaa a a a a a a a a x://foo.bar.test
```

generates links ... I think admins could easily misconfigure ...

Should we probably restrict protocols in the setting to only allow ASCII chars?